src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx

```diff
@@ -566,8 +573,8 @@ export class EditDataSourceForm extends React.Component<EditDataSourceProps, Edi
 
         {this.state.showUpdateAwsCredentialModal ? (
           <UpdateAwsCredentialModal
-            region={this.state.auth.credentials!.region}
-            service={this.state.auth.credentials!.service}
+            region={this.state.auth.credentials!.region?.toString() || ''}
+            service={this.state.auth.credentials!.service as SigV4ServiceName}
```

Comment:
These non-null assertion seems unnecessary
```suggestion
            region={this.state.auth.credentials.region?.toString() || ''}
            service={this.state.auth.credentials.service as SigV4ServiceName}
```